mmcv/transforms/wrappers.py

```diff
@@ -0,0 +1,307 @@
+                return None
+        return results
+
+    def __repr__(self):
+        """Compute the string representation."""
+        format_string = self.__class__.__name__ + '('
+        for t in self.transforms:
+            format_string += f'\n    {t}'
+        format_string += '\n)'
+        return format_string
+
+
+@TRANSFORMS.register_module()
+class Remap(BaseTransform):
+    """A transform wrapper to remap and reorganize the input/output of the
+    wrapped transforms (or sub-pipeline).
+
+    Args:
+        transforms (list[dict|callable]):
+        input_mapping (dict): A dict that defines the input key mapping.
+            The keys corresponds to the inner key (i.e. kwargs of the
+            `transform` method), and the values corresponds to the outer
+            keys (i.e. the keys of the data/results).
+        output_mapping(dict): A dict that defines the output key mapping.
+            The keys corresponds to the inner key (i.e. the keys of the
+            output dict of the `transform` method), and the values
+            corresponds to the outer keys (i.e. the keys of the
+            data/results).
+    """
+
+    def __init__(self,
+                 transforms: List[Union[Dict, Callable[[Dict], Dict]]],
+                 input_mapping: Optional[Dict] = None,
+                 output_mapping: Optional[Dict] = None,
+                 inplace: bool = False,
+                 strict: bool = True):
+
+        self.inplace = inplace
+        self.strict = strict
+        self.input_mapping = input_mapping
+
+        if inplace:
+            if not self.strict:
+                raise ValueError('Remap: `strict` must be set True if'
+                                 '`inplace` is set True.')
+
+            if output_mapping is not None:
+                raise ValueError('Remap: the output_mapping must be None '
+                                 'if `inplace` is set True.')
+            self.output_mapping = input_mapping
+        else:
+            self.output_mapping = output_mapping
+
+        self.transforms = Compose(transforms)
+
+    def remap_input(self, data: Dict, input_mapping: Dict) -> Dict[str, Any]:
+        """Remap inputs for the wrapped transforms by gathering and renaming
+        data items according to the input_mapping.
+
+        Args:
+            data (dict): The original data dictionary of the pipeline
+            input_mapping(dict):
+        """
+
+        def _remap(data, m):
+            if isinstance(m, dict):
+                # m is a dict {inner_key:outer_key, ...}
+                return {k_in: _remap(data, k_out) for k_in, k_out in m.items()}
+            if isinstance(m, (tuple, list)):
+                # m is a list [outer_key1, outer_key2, ...]
+                return m.__class__(_remap(data, e) for e in m)
+
+            # m is an outer_key
+            if self.strict:
+                return data.get(m)
+            else:
+                return data.get(m, NotInResults)
```

Comment:
Why do we provide an option that ignores the missing keys that are not in the `results`?

Reply:
It helps to allow more flexible use of the key mapping. For example, for transforms like Resize and Flip, a key mapping defined for aligned `img, bbox` can also handle the case where there is only `img`.
Do you think this option is necessary?

Reply:
It is necessary and should we print the warning log at the same time?